### PR TITLE
[R] update example of slice

### PR DESCRIPTION
**Pull request by thierrygosselin**
For the example to work I added `lightgbm::lgb.Dataset.construct(dsub)`
otherwise, it doesn't work

### CODE REVIEW

**Comment by thierrygosselin** [comment]
Quick question regarding to slice...
Maybe I'm not using it in the context deigned for, but is it normal that a sliced dataset can't be used after in a training session using `lightgbm::lgb.train` ?
I get this error:
```
Error in self$construct() : 
  lgb.Dataset.construct: does not support constructing from ‘lgb.Dataset’‘R6’
```

I wanted to subset my lgb.Dataset into a training and test set....

**Comment by Laurae2** [comment]
Going to check tomorrow.

**Comment by Laurae2** [comment]
@thierrygosselin Please review the following code. You do not need `lgb.Dataset.construct` as long as you don't need the LightGBM binary matrix built. Therefore, I do not know why the example does not work for you.

```r
library(lightgbm)
data(agaricus.train, package = "lightgbm")
train <- agaricus.train
dtrain <- lgb.Dataset(train$data, label = train$label)

dsub <- lightgbm::slice(dtrain, 1:42)
dsub$construct()
lightgbm::getinfo(dsub, "label")
nrow(dsub)

dsub2 <- dsub$slice(1:21)
dsub2$construct()
lightgbm::getinfo(dsub2, "label")
nrow(dsub2)
```

This is my output and it works fine:

```r
> library(lightgbm)
Loading required package: R6
> data(agaricus.train, package = "lightgbm")
> train <- agaricus.train
> dtrain <- lgb.Dataset(train$data, label = train$label)
> 
> dsub <- lightgbm::slice(dtrain, 1:42)
> dsub$construct()
Loading required package: Matrix
> lightgbm::getinfo(dsub, "label")
 [1] 1 0 0 1 0 0 0 1 0 0 0 0 0 0 0 1 1 1 0 1 0 0 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
> nrow(dsub)
[1] 42
> 
> dsub2 <- dsub$slice(1:21)
> dsub2$construct()
> lightgbm::getinfo(dsub2, "label")
 [1] 1 0 0 1 0 0 0 1 0 0 0 0 0 0 0 1 1 1 0 1 0
> nrow(dsub2)
[1] 21
```

I could even slice after constructing the LightGBM binary matrix (yes, the `getinfo` works only if the dataset is constructed when it comes from a sliced matrix).

Bigger example with training:

```r
library(lightgbm)
data(agaricus.train, package = "lightgbm")
params <- list(objective = "regression", metric = "l2")

train <- agaricus.train
dtrain <- lgb.Dataset(train$data, label = train$label)

dsub <- lightgbm::slice(dtrain, 1:42)
dsub$construct()
lightgbm::getinfo(dsub, "label")
nrow(dsub)

model <- lgb.train(params,
                   dsub,
                   2,
                   min_data = 1,
                   learning_rate = 1)

dsub2 <- dsub$slice(1:21)
dsub2$construct()
lightgbm::getinfo(dsub, "label")
nrow(dsub2)

model <- lgb.train(params,
                   dsub2,
                   2,
                   min_data = 1,
                   learning_rate = 1)
```
